src/bblocks/datacommons_tools/custom_data/data_management.py

```diff
@@ -869,26 +898,85 @@ def validate_config(self) -> CustomDataManager:
         self._config.validate_config()
         return self
 
-    def __repr__(self) -> str:
-        input_files_count = len(self._config.inputFiles)
-        sources_count = len(self._config.sources)
-        nodes_count = sum(
-            [len(var) for var in [n.nodes for n in self._mcf_nodes.values()] if var]
-        )
+    def merge_config(
```

Comment:
I wonder if the name of the method should instead be `add_config` because what we are actually doing is an outer merge more akin to appending

Reply:
That actually depends on the policy because you can override